tensorboard/webapp/metrics/store/metrics_selectors.ts

```diff
@@ -472,3 +476,45 @@ export const isMetricsSlideoutMenuOpen = createSelector(
   selectMetricsState,
   (state): boolean => state.isSlideoutMenuOpen
 );
+
+/**
+ * Gets the min and max step of a metrics card.
+ */
+export const getMetricsCardMinMax = createSelector(
+  getCardStateMap,
+  (cardStateMap: CardStateMap, cardId: CardId): MinMaxStep | undefined => {
+    return getMinMaxStepFromCardState(cardStateMap[cardId]);
+  }
+);
+
+/**
+ * Gets the time selection of a metrics card.
+ */
+export const getMetricsCardTimeSelection = createSelector(
+  getCardStateMap,
+  getMetricsLinkedTimeEnabled,
+  getMetricsLinkedTimeSelection,
+  (
+    cardStateMap: CardStateMap,
+    linkedTimeEnabled: boolean,
+    linkedTimeSelection: TimeSelection | null,
+    cardId: CardId
+  ): TimeSelection | undefined => {
+    if (linkedTimeEnabled && linkedTimeSelection) {
```

Comment:
This could technically just rely on the `linkedTimeSelection` being non-null because `getMetricsLinkedTimeSelection` returns `null` when linked time is not enabled. However, don't think that is obvious to a reader so I added this null check. I could imagine a comment having a similar effect.